crates/sawtooth-tp/src/tp.rs

```diff
@@ -216,6 +255,62 @@ impl TP for ChronicleTransactionHandler {
 
         Ok(effects)
     }
+
+    /// Get identity, the content of the compact json-ld representation of the operation, and a more
+    /// readable serialization of the @graph from a provmodel containing the operation's dependencies and then
+    /// pass them as the context to an OPA rule check, returning an error upon OPA policy failure
+    async fn enforce_opa(
+        opa_executor: &ExecutorContext,
+        identity: &SignedIdentity,
+        tx: &ChronicleOperation,
+        state: &OperationState<SawtoothAddress>,
+    ) -> Result<(), ApplyError> {
+        let identity = AuthId::try_from(identity)
+            .map_err(|e| ApplyError::InternalError(ProcessorError::SerdeJson(e).to_string()))?;
+
+        let identity_context = identity
+            .identity_context()
+            .map_err(|e| ApplyError::InternalError(e.to_string()))?;
+
+        // Set up Context for OPA rule check
+        let operation = serde_json::to_value(
+            tx.to_json()
+                .compact()
+                .await
+                .map_err(|e| ApplyError::InternalError(ProcessorError::Compaction(e).to_string()))?
+                .dump(),
+        )
+        .map_err(|e| ApplyError::InternalError(ProcessorError::SerdeJson(e).to_string()))?;
+
+        // Get the dependencies
+        let deps = tx
+            .dependencies()
+            .into_iter()
+            .collect::<HashSet<_>>()
+            .iter()
+            .map(SawtoothAddress::from)
+            .collect::<HashSet<_>>();
+
+        let operation_state = state.opa_context(deps);
+
+        let state = tx
+            .opa_context_state(ProvModel::default(), operation_state)
+            .await
+            .map_err(|e| ApplyError::InternalError(e.to_string()))?;
+
+        let context = serde_json::json!({
+            "identity": identity_context,
+            "operation": operation,
+            "state": state,
+        });
+
+        info!(opa_evaluation_context = %context);
+
+        match opa_executor.evaluate(&identity, &context).await {
+            Ok(_) => Ok(()),
```

Comment:
I am most curious that an `Ok(false)` is just fine here, we're just checking that we _can_ evaluate it.

Reply:
Thanks for asking about that. The updated `evaluate` method returns an error if the result of the evaluation is false, so an `Ok` `Result` here will always be `Ok(true)`. Any suggestions on how to make that less confusing?

Reply:
I've added a comment for clarification.

Reply:
I'd suggest returning `Ok(())` - the `bool`'s become superfluous so that's why it's confusing.